x/accounts/internal/implementation/protoaccount.go

```diff
@@ -0,0 +1,55 @@
+package implementation
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+// ProtoMsg is a generic interface for protobuf messages.
+type ProtoMsg[T any] interface {
```

Comment:
aaron mentioned protov2 doesnt work with the sdk yet. does that carry over to this?

Reply:
from SDK point of view this can be JSON bytes and accounts will convert JSON bytes to a protov2 message. 
I've structured accounts to also support a different encoding (which can be gogo), I didn't do it here because it'd be a lot more complex to deal with. 

I'll make you an example on how this works on the SDK using protov1:

```
accounts.MsgExecute{
     Sender: addr,
     Target: x/accounts_account_addr
     Message: `{"type": "protov2.message", "value": {"some execution data"}`
}
```

So the above message is protov1 (gogo) and is understood by the SDK, the protov2 message is JSONified and is bytes (it could also be proto-bytes). Then when the json bytes come to x/accounts, the module is smart enough to know how to convert that into a concrete protov2 message.

Reply:
This is only important at TX level, in case of communication between module and account, the module should just send the protov2 message using the accounts router and will be able to talk with the target.